src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -203,42 +204,45 @@ sealed class HashMap[A, +B] extends AbstractMap[A, B]
     if (isCompatibleCBF(bf)) addSimple(that)
     else super.++:(that)
   }
-  private def addSimple[C >: (A, B), That](that: TraversableOnce[C]): That = {
+  private def addSimple[C >: (A, B), That](that: TraversableOnce[C])(implicit bf: CanBuildFrom[HashMap[A, B], C, That]): That = {
```

Comment:
can we drop the `cbf`. Its only to make the cast hidden

Reply:
It drives inference of `That`. JIT will inline this into oblivion.